aws/src/main/java/org/apache/iceberg/aws/s3/S3InputStream.java

```diff
@@ -165,7 +165,9 @@ public void readFully(long position, byte[] buffer, int offset, int length) thro
 
     String range = String.format("bytes=%s-%s", position, position + length - 1);
 
-    IOUtil.readFully(readRange(range), buffer, offset, length);
+    try (InputStream stream = readRange(range)) {
```

Comment:
`readTail` method has the same problem no?

Reply:
Nice find @nandorKollar (bad assumption that `readFully` would close the stream).  I would agree there's the same issue for `readTail` and the other implementations.